### PR TITLE
feat(backend): add error message if loculus accessions are duplicated…

### PR DESCRIPTION
**Pull request by anna-parker**
… in revision file - actually add some unit tests for revision (there are currently none!)

resolves https://github.com/loculus-project/loculus/issues/5248

### PR Checklist
- [ ] All necessary documentation has been adapted.
- [ ] The implemented feature is covered by appropriate, automated tests.
- [ ] Any manual testing that has been done is documented (i.e. what exactly was tested?)

🚀 Preview: Add `preview` label to enable

### CODE REVIEW

**Comment by anna-parker** [comment]
> There are various tests here fwiw, maybe you don't consider them unit tests https://github.com/loculus-project/loculus/blob/a67ea9f129c27366b1fbf791c4d068a92e1e054a/backend/src/test/kotlin/org/loculus/backend/controller/submission/ReviseEndpointTest.kt

A crap I didn't see them... so I just wasted an hour reproducing them - it is a bit weirdly structured as there are no tests for the UploadDatabaseService just the controllers. I will move my tests there though for consistency.

File: backend/src/main/kotlin/org/loculus/backend/service/submission/UploadDatabaseService.kt [hunk 1]
```diff
@@ -269,31 +271,44 @@ class UploadDatabaseService(
...
-                AND ${SequenceEntriesTable.isMaxVersion}
-        """.trimIndent()
         transaction {
-            exec(
-                updateSql,
-                listOf(
-                    Pair(VarCharColumnType(), uploadId),
-                ),
-            )
+            val rows = SequenceEntriesTable
```
[the remark this answers is not in the document]
Comment (anna-parker) [reply backend/src/main/kotlin/org/loculus/backend/service/submission/UploadDatabaseService.kt]: In jooq this should be possible with a very simple (almost SQL like expression):

val m = METADATA_UPLOAD_AUX_TABLE
val s = SEQUENCE_ENTRIES

dsl.update(m)
    .set(m.VERSION, s.VERSION.plus(1))
    .set(m.GROUP_ID, s.GROUP_ID)
    .from(s)
    .where(
        m.UPLOAD_ID.eq(uploadId)
            .and(m.ACCESSION.eq(s.ACCESSION))
            .and(SequenceEntriesTable.IS_MAX_VERSION)
    )
    .execute()

File: backend/src/test/kotlin/org/loculus/backend/controller/submission/ReviseEndpointTest.kt [hunk 2]
```diff
@@ -155,6 +155,31 @@ class ReviseEndpointTest(
             )
     }
 
+    @Test
```
Comment (corneliusroemer) [inline backend/src/test/kotlin/org/loculus/backend/controller/submission/ReviseEndpointTest.kt]: To test the chunk issue, we might want to add another test that sets chunk size to n=1 :)

**Review by corneliusroemer — CHANGES_REQUESTED** [review]
Good stuff thanks! I think we still need to deal with the chunking issue though, see comments
